addons/l10n_id_efaktur_coretax/models/account_move.py

```diff
@@ -0,0 +1,317 @@
+            raise RedirectWarning(msg, action_error, _("Display Related Documents"))
+
+        return self.download_xml()
+
+    def download_xml(self):
+        return self.l10n_id_efaktur_document.action_download()
+
+    def prepare_efaktur_vals(self):
+        """ Get information required from invoice and lines to generate E-Faktur that will be used
+        to load in the XML template later on"""
+        invoice_vals = []
+        idr = self.env.ref('base.IDR')
+
+        for move in self.filtered(lambda m: m.state == 'posted'):
+
+            commercial_partner = move.partner_id.commercial_partner_id
+            trx_code = move.l10n_id_kode_transaksi
+
+            vals = {
+                "TIN": move.company_id.vat,
+                "TaxInvoiceDate": move.invoice_date.strftime("%Y-%m-%d"),
+                "TaxInvoiceOpt": "Normal",
+                "TrxCode": trx_code,
+                "AddInfo": move['l10n_id_add_info_' + trx_code] if trx_code in ('07', '08') else "",
+                "CustomDoc": "",
+                "FacilityStamp": move['l10n_id_facility_info_' + trx_code] if trx_code in ('07, 08') else "",
+                "RefDesc": move.name,
+                "SellerIDTKU": move.company_id.vat + move.company_id.partner_id.l10n_id_tku,
+                "BuyerDocument": commercial_partner.l10n_id_buyer_document_type,
+                "BuyerTin": commercial_partner.vat if commercial_partner.l10n_id_buyer_document_type == "TIN" else "0000000000000000",
+                "BuyerCountry": COUNTRY_CODE_MAP.get(commercial_partner.country_id.code),
+                "BuyerDocumentNumber": commercial_partner.l10n_id_buyer_document_number if commercial_partner.l10n_id_buyer_document_type != "TIN" else "",
+                "BuyerName": commercial_partner.name,
+                "BuyerAdress": commercial_partner.contact_address.replace('\n', ' ').strip(),
+                "BuyerEmail": commercial_partner.email or "",
+                "BuyerIDTKU": commercial_partner.vat + commercial_partner.l10n_id_tku,
+            }
+
+            # vals["lines"] = move.invoice_line_ids._prepare_efaktur_vals()
+            line_vals = []
+            trx_code = move.l10n_id_kode_transaksi
+
+            for line in move.invoice_line_ids:
+                product = line.product_id
+
+                line_val = {
+                    "Opt": "B" if product.detailed_type == "service" else "A",  # A: goods, B: service
+                    "Code": product.l10n_id_product_code.code or self.env.ref('l10n_id_efaktur_coretax.product_code_000000_goods').code,
+                    "Name": product.name,
+                    "Unit": line.product_uom_id.l10n_id_uom_code.code,
+                    "Price": idr.round(line.price_unit),
+                    "Qty": line.quantity,
+                    "TotalDiscount": idr.round(line.discount * line.quantity * line.price_unit / 100),
+                    "TaxBase": idr.round(line.price_subtotal),  # DPP
```

Comment:
You don't need to do the workaround here?

Reply:
So we use the "OtherTaxBase" now where it's multiplied by 11/12 ratio